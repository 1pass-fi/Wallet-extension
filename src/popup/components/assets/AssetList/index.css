.assets-add-more-row {
  background-color: #EEE;
  height: 54px;
  width: 100%;
  display: flex;
  align-items: center;
}

.assets-plus-icon {
  margin-left: 35px;
  width: 24px;
  height: 24px;
  cursor: pointer;
}