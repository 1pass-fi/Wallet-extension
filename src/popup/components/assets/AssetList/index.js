import React, { useState, useEffect } from 'react'
import { TYPE } from 'account/accountConstants' 
import PropTypes from 'prop-types'

import AssetRow from '../AssetRow'
import PlusIcon from 'img/plus-icon.svg'
import './index.css'
import CollapseIcon from 'img/collapse-icon.svg'
import ExtendIcon from 'img/extend-icon.svg'

import { popupAccount } from 'account'

const propTypes = {
  assets: PropTypes.array,
  onAddAsset: PropTypes.func,
}

const AssetList = ({ owner, assets, onAddAsset }) => {
  const [ownerName, setOwnerName] = useState(null)
  const [collapsed, setCollapsed] = useState(false)

  useEffect(() => {
    const getOwnerName = async () => {
      console.log('owner: ', owner)
      const account = await popupAccount.getAccount({ address: owner })
      const name = await account.get.accountName()
      console.log(name)
      setOwnerName(name)
    }

    getOwnerName()
  }, [])

  return (
    <div className={collapsed ? 'assets collapsed' : 'assets'}>
      <div className="owner">
        {ownerName}
        {!collapsed && <div onClick={() => setCollapsed(!collapsed)} className="collapse-icon"><CollapseIcon /></div>}
        {collapsed && <div onClick={() => setCollapsed(!collapsed)} className="collapse-icon"><ExtendIcon /></div>}
      </div>
      {assets.map((asset, index) => <AssetRow key={index} {...asset} isGrey={index % 2 === 0} />)}
      <div className="assets-add-more-row">
        <div onClick={onAddAsset} className="assets-plus-icon">
          <PlusIcon />
        </div>
      </div>
    </div>
  )
}

AssetList.propTypes = propTypes

export default AssetList
