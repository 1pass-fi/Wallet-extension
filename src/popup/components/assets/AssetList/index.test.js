import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import React from 'react'

import AssetList from '.'

const assets = [
  {
    isKoiWallet: true,
    name: 'Astropinx',
    isRegistered: true,
    earnedKoi: 29.373
  }, 
  {
    isKoiWallet: false,
    name: 'AsMercury (The Planets #1)tropinx',
    isRegistered: true,
    earnedKoi: 29.373
  }, 
]

describe('Test for AssetList component', () => {
  describe('Render without crashing', () => {
    it('renders correctly', () => {
      const { container } = render(<AssetList assets={assets} />)
      expect(container).toMatchSnapshot()
    })
  })
})
