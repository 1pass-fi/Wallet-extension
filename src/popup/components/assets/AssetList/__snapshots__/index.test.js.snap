// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for AssetList component Render without crashing renders correctly 1`] = `
<div>
  <div
    class="assets"
  >
    <div
      class="asset-row-container"
      style="background: rgb(238, 238, 238);"
    >
      <div
        class="asset-row-rearrange-icon"
      >
        <svg />
      </div>
      <div
        class="asset-row-logo-icon"
      >
        <svg />
      </div>
      <div
        class="asset-name"
      >
        Astropinx
      </div>
      <div
        class="koi-earned"
      >
        29.373
         KOI earned
      </div>
      <div
        class="asset-row-function-icons"
      >
        <div>
          <svg
            class="asset-row-function-icon"
          />
        </div>
        <div>
          <svg
            class="asset-row-function-icon"
          />
        </div>
      </div>
    </div>
    <div
      class="asset-row-container"
      style="background: rgb(255, 255, 255);"
    >
      <div
        class="asset-row-rearrange-icon"
      >
        <svg />
      </div>
      <div
        class="asset-row-logo-icon"
      >
        <svg />
      </div>
      <div
        class="asset-name"
      >
        AsMercury (The Planets #1)tropinx
      </div>
      <div
        class="koi-earned"
      >
        29.373
         KOI earned
      </div>
      <div
        class="asset-row-function-icons"
      >
        <div>
          <svg
            class="asset-row-function-icon"
          />
        </div>
        <div>
          <svg
            class="asset-row-function-icon"
          />
        </div>
      </div>
    </div>
    <div
      class="assets-add-more-row"
    >
      <div
        class="assets-plus-icon"
      >
        <svg />
      </div>
    </div>
  </div>
</div>
`;
